include/util/struct_packing.hpp

```diff
@@ -0,0 +1,19 @@
+#ifndef STRUCT_PACKING_HPP
+#define STRUCT_PACKING_HPP
+
+/*
+    Macro to pack struct/class in both environment
+    Usage:  PACK( 
+        struct MyStruct 
+        { 
+            int Property1;
+        });
+*/
+
+#ifdef _MSC_VER
+#define PACK( input ) __pragma( pack(push, 1) ) input __pragma( pack(pop) )
```

Comment:
@TheMarex , when you said struct was not packable, do you mean when struct alignment in such manner there's no different in packed or default behavior? (aka sizeof is the same packed or not).  If that's the case please check this out under gcc 6.1, it doesn't look like it complaining either way:  http://melpon.org/wandbox/permlink/S8O81vF5WLqGHhIM